app/components/click_observer_component.rb

```diff
@@ -1,14 +1,21 @@
 # frozen_string_literal: true
 
 class ClickObserverComponent < BaseComponent
-  attr_reader :event_name, :tag_options
+  attr_reader :event_name, :tag_options, :payload
 
-  def initialize(event_name:, **tag_options)
+  def initialize(event_name:, payload: nil, **tag_options)
     @event_name = event_name
+    @payload = payload
     @tag_options = tag_options
   end
 
   def call
-    content_tag(:'lg-click-observer', content, 'event-name': @event_name, **tag_options)
+    content_tag(
+      :'lg-click-observer',
+      content,
+      'event-name': @event_name,
+      data: { payload: payload },
```

Comment:
Nit / optional: Since we're already using custom attributes without `data-` prefixes, I think we could be consistent one way or the other (`event-name="..." payload="..."` or `data-event-name="..." data-payload="..."`)

Reply:
good point updating.